baseplate/frameworks/thrift/__init__.py

```diff
@@ -103,6 +103,11 @@ def call_processor_with_span_context(
             if edge_context_factory:
                 context.edge_context = edge_context_factory.from_upstream(edge_payload)
 
+            deadline_budget = headers.get(b"Deadline-Budget", None)
+            context.deadline_budget = (
+                float(deadline_budget.decode()) / 1000 if deadline_budget else None
+            )
```

Comment:
```suggestion
            try:
                raw_deadline_budget = headers[b"Deadline-Budget"].decode()
                context.deadline_budget = float(raw_deadline_budget) / 1000
            except (KeyError, ValueError):
                context.deadline_budget = None
```

It'd be good to do some error handling so we can't accidentally end up crashing apps by sending e.g `Deadline-Budget: hahahah`.  Just for example purposes I restructured this as a big try/except since I think making the code less hedgy  (`.get()` etc) makes it a bit clearer. But feel free to do it elsehow if this doesn't make sense.

Reply:
this is a great point, thanks!